lib/Net/Async/Blockchain/Omni.pm

```diff
@@ -129,16 +128,18 @@ async sub transform_transaction {
     catch {
         # transaction not found
         return undef;
-    };
+    }
 
     # transaction not found, just ignore.
     return undef unless $received_transaction && $received_transaction->{ismine};
 
     my $amount = Math::BigFloat->new($received_transaction->{amount});
     my $fee = Math::BigFloat->new($received_transaction->{fee} // 0);
 
-    my $from = await $self->rpc_client->validate_address($received_transaction->{sendingaddress});
-    my $to   = await $self->rpc_client->validate_address($received_transaction->{referenceaddress});
+    my ($from, $to) = await Future->needs_all(
+        $self->rpc_client->validate_address($received_transaction->{sendingaddress}),
```

Comment:
since it's just the hash key that's different on these two lines, maybe worth a `map`?

Reply:
Good call. Done 9813149